### PR TITLE
Add Nuvoton NPCM400 SMC initial support

### PR DESCRIPTION
**Pull request by maxdog988**
The Nuvoton NPCM400 is a Satellite Management Controller for a wide range of server applications.

- This PR includes three commits for the initial bring up:
1. soc: arm: add nuvoton npcm400 support
2. boards: arm: add npcm400 evb board
3. MAINTAINERS: Add Nuvoton NPCM Platform for Nuvoton NPCM family

### CODE REVIEW

**Comment by fabiobaltieri** [comment]
Neat, took me a bit to figure whether this was going in a datacenter or in space. :-)

Looks like twister is unhappy about something, maybe something off in the soc or board definition, not too sure but can you try to reproduce it locally?

More importantly, does this need an update of MAINTAINERS.yml? https://github.com/zephyrproject-rtos/zephyr/blob/main/MAINTAINERS.yml#L3364 If it's a separate team for you folks it's probably worth adding a new area, if it's not than the file filter should be adjusted.

File: boards/nuvoton/npcm400f_evb/Kconfig.npcm400f_evb [hunk 1]
```diff
@@ -0,0 +1,6 @@
+# Copyright (c) 2024 Nuvoton Technology Corporation.
+# SPDX-License-Identifier: Apache-2.0
+
+config BOARD_NPCM400F_EVB
+	bool "Nuvoton NPCM400F EVB Development board"
```
Comment (nordicjm) [inline boards/nuvoton/npcm400f_evb/Kconfig.npcm400f_evb:5]: this is in hwmv1

**Review by nordicjm — DISMISSED** [review]

File: boards/nuvoton/npcm400f_evb/doc/npcm400f_evb.jpg
Comment (nordicjm) [inline boards/nuvoton/npcm400f_evb/doc/npcm400f_evb.jpg:1]: convert to webp then put through https://tinypng.com/

File: soc/nuvoton/npcm/Kconfig [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+# Copyright (c) 2024 Nuvoton Technology Corporation.
+# SPDX-License-Identifier: Apache-2.0
+
+if SOC_FAMILY_NPCM
+
+endif # SOC_FAMILY_NPCM
+
+# Select SoC Part No. and configuration options
+rsource "*/Kconfig"
```
Comment (nordicjm) [inline soc/nuvoton/npcm/Kconfig:11]: this goes in the if block above

File: soc/nuvoton/npcm/Kconfig.soc [hunk 3]
```diff
@@ -0,0 +1,12 @@
+# Nuvoton Cortex-M4 Embedded Controller
+
+# Copyright (c) 2024 Nuvoton Technology Corporation.
+# SPDX-License-Identifier: Apache-2.0
+
+config SOC_FAMILY_NPCM
+	bool
+
+config SOC_FAMILY
+	default "nuvoton_npcm" if SOC_FAMILY_NPCM
```
Comment (nordicjm) [inline soc/nuvoton/npcm/Kconfig.soc:10]:
```suggestion
	default "npcm" if SOC_FAMILY_NPCM
```
or
```suggestion
	default "nuvoton_npcm" if SOC_FAMILY_NUVOTON_NPCM
```
If the second then soc.yml needs update also

File: soc/nuvoton/npcm/npcm4/Kconfig [hunk 4]
```diff
@@ -0,0 +1,14 @@
...
+
+config SOC_SERIES_NPCM4
+	select ARM
+	select CPU_CORTEX_M4
+	select CPU_CORTEX_M_HAS_DWT
+	select SOC_FAMILY_NPCM
+	select CORTEX_M_SYSTICK
+	select CPU_HAS_ARM_MPU
+	help
+	  Enable support for Nuvoton NPCM4 series
```
Comment (nordicjm) [inline soc/nuvoton/npcm/npcm4/Kconfig:14]:
```suggestion
```

File: soc/nuvoton/npcm/npcm4/soc.c [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+
+#include <zephyr/kernel.h>
+#include <zephyr/device.h>
+#include <zephyr/init.h>
+#include <soc.h>
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_REGISTER(soc, CONFIG_SOC_LOG_LEVEL);
+
+static int soc_npcm4_init(void)
```
Comment (nordicjm) [inline soc/nuvoton/npcm/npcm4/soc.c:15]: this whole file does nothing and can go

File: soc/nuvoton/npcm/common/CMakeLists.txt [hunk 6]
```diff
@@ -0,0 +1,16 @@
+#
+# Copyright (c) 2024 Nuvoton Technology Corporation.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+zephyr_include_directories(.)
+
+set(NPCM_BIN_NAME ${CONFIG_KERNEL_BIN_NAME}_signed.bin)
```
Comment (nordicjm) [inline soc/nuvoton/npcm/common/CMakeLists.txt:8]: this is not signed, it has a header added and basically copies the same filename as mcuboot-signed images

**Comment by maxdog988** [comment]
> Neat, took me a bit to figure whether this was going in a datacenter or in space. :-)
> 
> Looks like twister is unhappy about something, maybe something off in the soc or board definition, not too sure but can you try to reproduce it locally?
> 
> More importantly, does this need an update of MAINTAINERS.yml? https://github.com/zephyrproject-rtos/zephyr/blob/main/MAINTAINERS.yml#L3364 If it's a separate team for you folks it's probably worth adding a new area, if it's not than the file filter should be adjusted.

File: boards/nuvoton/npcm400f_evb/Kconfig.npcm400f_evb (hunk 1 above)
Comment (maxdog988) [reply boards/nuvoton/npcm400f_evb/Kconfig.npcm400f_evb]: Thanks for reviewing!  Resolved

File: boards/nuvoton/npcm400f_evb/doc/npcm400f_evb.jpg
Comment (maxdog988) [reply boards/nuvoton/npcm400f_evb/doc/npcm400f_evb.jpg:1]: Thanks for reviewing!  Resolved

File: soc/nuvoton/npcm/Kconfig (hunk 2 above)
Comment (maxdog988) [reply soc/nuvoton/npcm/Kconfig:7]: Thanks for reviewing!  Resolved

File: soc/nuvoton/npcm/Kconfig.soc (hunk 3 above)
Comment (maxdog988) [reply soc/nuvoton/npcm/Kconfig.soc]: Go with npcm!
Thanks for reviewing! Resolved

File: soc/nuvoton/npcm/npcm4/Kconfig (hunk 4 above)
Comment (maxdog988) [reply soc/nuvoton/npcm/npcm4/Kconfig]: Thanks for reviewing! Resolved

File: soc/nuvoton/npcm/npcm4/soc.c (hunk 5 above)
Comment (maxdog988) [reply soc/nuvoton/npcm/npcm4/soc.c:15]: Thanks for reviewing! Resolved

File: soc/nuvoton/npcm/common/CMakeLists.txt (hunk 6 above)
Comment (maxdog988) [reply soc/nuvoton/npcm/common/CMakeLists.txt]: Thanks for reviewing!
Yes, this is not signed, so I changed to .npcm.bin.

**Comment by maxdog988** [comment]
> Neat, took me a bit to figure whether this was going in a datacenter or in space. :-)
>
Thanks for reviewing. 
> Looks like twister is unhappy about something, maybe something off in the soc or board definition, not too sure but can you try to reproduce it locally?
> 
Got it! I will try to reproduce it locally.
> More importantly, does this need an update of MAINTAINERS.yml? https://github.com/zephyrproject-rtos/zephyr/blob/main/MAINTAINERS.yml#L3364 If it's a separate team for you folks it's probably worth adding a new area, if it's not than the file filter should be adjusted.
> 
Yes, we are different team, I have added a MAINTAINERS commit in this PR.

**Review by fabiobaltieri — DISMISSED** [review]

**Comment by maxdog988** [comment]
Updated this PR to change the soc name from NPCM400F to NPCM400.

**Review by fabiobaltieri — APPROVED** [review]

**Review by MulinChao — APPROVED** [review]

**Pull request merged**